protocols/imap/src/main/java/org/apache/james/imap/main/PathConverter.java

```diff
@@ -38,41 +45,48 @@ public PathConverter(ImapSession session) {
     }
 
     public MailboxPath buildFullPath(String mailboxName) {
-        String namespace = null;
-        String name = null;
-        final MailboxSession mailboxSession = ImapSessionUtils.getMailboxSession(session);
-
-        if (mailboxName == null || mailboxName.length() == 0) {
+        if (Strings.isNullOrEmpty(mailboxName)) {
             return new MailboxPath("", "", "");
         }
-        if (mailboxName.charAt(0) == MailboxConstants.NAMESPACE_PREFIX_CHAR) {
-            int namespaceLength = mailboxName.indexOf(mailboxSession.getPathDelimiter());
-            if (namespaceLength > -1) {
-                namespace = mailboxName.substring(0, namespaceLength);
-                if (mailboxName.length() > namespaceLength)
-                    name = mailboxName.substring(++namespaceLength);
-            } else {
-                namespace = mailboxName;
-            }
+        if (isAbsolute(mailboxName)) {
+            return buildAbsolutePath(mailboxName);
         } else {
-            namespace = MailboxConstants.USER_NAMESPACE;
-            name = mailboxName;
+            return buildRelativePath(mailboxName);
         }
-        String user = null;
-        // we only use the user as part of the MailboxPath if its a private user
-        // namespace
+    }
+
+    private boolean isAbsolute(String mailboxName) {
```

Comment:
Absolute path have a namespace specified.

Relative path concerns user mailboxes.

Hence the check to namespace char.